NewArch/src/Navigation.tsx

```diff
@@ -42,8 +42,36 @@ type RouteType = {
   params: any,
 }
 
-type NavigationContainerProps = PropsWithChildren<{}>;
-const NavigationContainer = ({children}: NavigationContainerProps) => {
+type Theme = {
+  dark: boolean;
+  colors: {
+    primary: string;
+    background: string;
+    card: string;
+    text: string;
+    border: string;
+    notification: string;
+  };
+};
+
+type NavigationContainerProps = PropsWithChildren<{
+  theme?: Theme;
+}>;
```

Comment:
I'm confused about the mix of a Context-style theme, and a pass through by prop theme. All while App.tsx has two other Theme contexts. Why the change to theme design here? These seems like more and I'm not sure why it doesn't blend with what's already there. Can you clarify the design?

Reply:
But why are you creating new theme objects and types when there are already ones in src\themes and also referred to in this app's App.tsx?

Reply:
Yes, thank you for tracking and documenting.